pkg/crawler/log.go

```diff
@@ -56,3 +72,97 @@ func escapeString(str string) string {
 	}
 	return string(b)
 }
+
+// Check File path and create if not exists
+func (c *Crawler) checkFileAndCreate(filePath string) {
+	// Check directory and create if not exists
+	dir := filepath.Dir(filePath)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		err := os.MkdirAll(dir, 0755)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+	// Check file and create if not exists
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		_, err := os.Create(filePath)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+// Write files to the output writer
+func (c *Crawler) writeLineToFIle(message, filePath string) {
+	// filePath check and create
+	c.checkFileAndCreate(filePath)
+
+	file, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+
+	var error string
+	var status float64
+	var url string
+	var time float64
+	var duration float64
+
+	var data map[string]interface{}
+	if err := json.Unmarshal([]byte(message), &data); err != nil {
+		log.Fatal(err)
+	}
+
+	if reflect.TypeOf(data["url"]) != nil || reflect.TypeOf(data["time"]) != nil || reflect.TypeOf(data["duration"]) != nil {
```

Comment:
I'm not quite sure why this is necessary, you presumably are checking for existence right? In that case in Go we would usually do:

```go
url, ok := data["url"]
if !ok {
    // ...
}
```

Although for this example I feel like it would possibly be better to just create an anonymous struct

```go
var data struct {
    err string
    status int
    // ...
}
json.Unmarshal([]byte(message), &data)
```

in that case the usage of the uninitialized variables disappears completely and you also dont have to cast them around

I also don't quite get why you cast them into float64 only to then cast them back into int

Reply:
Yes, it throws an error that stops the program if there is no data.

I was getting an error when I sent int type data. That's why I was giving `float64` value and then converting it back to `int`. I solved that problem as you said.